packages/runtime/src/useCases/consumption/attributes/ShareRepositoryAttribute.ts

```diff
@@ -54,7 +54,8 @@ export class ShareRepositoryAttributeUseCase extends UseCase<ShareRepositoryAttr
             "content.owner": this.accountController.identity.address.toString(),
             "content.@type": "IdentityAttribute",
             "shareInfo.sourceAttribute": request.attributeId,
-            "shareInfo.peer": request.peer
+            "shareInfo.peer": request.peer,
+            "deletionInfo.deletionStatus": { $ne: DeletionStatus.DeletedByPeer }
```

Comment:
I am not in favor of doing this, since the point is to avoid confusion from having multiple shared Attributes with the same value. If one if these Attributes is now marked for deletion and another isn't, the user won't understand this. Also, I think the case you are describing is an absolute edge case (why would you want to share an Attribute with somebody you just requested its deletion from?). And wanting to do so, you can just share the Attribute with them again as soon as they deleted their Attribute, such that you and the peer both have neat distinguished Attributes. So, for the time being I'd say let's not implement this and if a request for this feature arises we can discuss this again.

Reply:
I see it differently now that I have thought about it again. Blocking the user from sharing an Attribute again seems to me to be a greater disadvantage compared to the confusion that may arise if there are temporarily (during the deletion period of the Attribute) two Attributes with the same value. I have thought of two scenarios:
- Scenario 1: I have shared a phone number and email address with someone. Now, as a knee-jerk reaction, I request the deletion of my email address because I assume at this moment that my email address is no longer needed and data security is important to me. In the next moment, however, I realize that the peer has a reason to have my email address and I would like to be able to undo the action or at least share my email address again if, for example, I know that I am not easy or pleasant to contact via my telephone number. Something like this has already happened to me and it frustrated me to be tied to processes with a deterministic (!) outcome. 
- Scenario 2: I have shared my email address and requested its deletion because I think it is no longer needed. Then, for example, I make a purchase from the peer and choose a payment option for which the peer requires my email address. Now this peer may not have completed the deletion of my old email address yet, but is no longer using it (depending on how these processes are defined at the peer's side) because it is in the `DeletionStatus.ToBeDeleted` status and the deletion date may is in the near future before the customer's purchase is completed, which is why I receive a Request for my email address from the peer. I cannot then reply to this Request with this email address, even if I would then perhaps consider it appropriate to share this email address again after the need to do so arises.
Long story short: I can see use cases where it is needed to re-share an Attribute in status `DeletionStatus.ToBeDeleted` and I consider the disadvantage of confusion due to duplicate Attributes to be less, because this state only lasts temporarily and requires a conscious action by the user to create this situation of duplicate Attributes. And in case this was unintentionally caused, a deletion could be requested again for these duplicate Attributes.

Reply:
Your scenarios bear upon the premise that an attribute whose deletion has been requested but not yet executed is not properly usable. And - unless I'm missing a detail - that's wrong since the attribute is still available to the peer (i.e. the peer _has_ access to your email address).

>Now this peer may not have completed the deletion of my old email address yet, but is no longer using it

That would violate our gentleman's agreement around which the deletion of attributes was designed. This would be a user error, which I'd rather prevent with clear and understandable documentation (which _necessitates_ simple logic if we want the user to really grok it) than to expand our already very complex attribute logic with additional paths and safety nets so that neither we nor the user can ever hope to have a clear and simple mental model of whats going on.

**tl;dr** KISS YAGNI HDGDL

Reply:
The Attribute can continue to be used in status `DeletionStatus.ToBeDeleted` (but does not have to for new transactions), but in all cases only until the deletion date. I meant that it could be that the email address has, for example, the deletion date of June 26th, but I have now ordered something that requires a longer existing email address. It is inconvenient to have a shared Attribute that will be deleted at the peer's side because its validity is limited and there will therefore be difficulties with transactions if it is already foreseeable that a transaction will soon be initiated. In this case, I cannot anticipate and decide on my own to share the Attribute with unlimited validity, but am bound to the fact that the Attribute must be deleted first. As I said, I know such problems from my own user experience and found them annoying. :D It also means no implementation effort, apart from including this status in the IF condition, too. Therefore, I think that the main counterargument would be Attribute duplication, but I would give more weight to the more problem-free functionality of the scenarios described above.

**tl;dr** IMHO adding the `DeletionStatus.ToBeDeleted` here is 👜 LIGHT LUGGAGE 👜 for our code base. 😂 

@Milena-Czierlinski I have expressed my opinion on this. Please think about it and then decide. ✨

Reply:
@britsta Just because I opened the PR doesn't mean this functional decision should be made by me alone. I hope we can reach an agreement everybody feels is suitable. :)

I agree with you that implementation-wise this would be no problem, but that doesn't affect the question, whether we want to have it functionality-wise. I also understand your example and it is hard for me to balance which case will occur more often. Having the user request the deletion of one of their attributes even though it is needed for certain processes, having the peer accept the deletion and naming a deletion date in the future that is far enough for the user to trigger the process requiring the attribute and then taking the process so long that it cannot be finished before the deletion date is reached seems like rather an edge case to me. But if this should occur, it wouldn't be optimal for the user to create a second repository attribute or for the peer to first having to delete the attribute before asking for it again. On the other hand we have the topic of attribute duplication as already discussed. This also feels awkward and I can already hear UX complaining.

@jkoenig134 Do you have a strong opinion on this?

Reply:
I don't.

Reply:
This would be a rejection of the deletion process which we don't have, right? I think we won't implement something like this and so I'm in favor to let it so like Milena has implemented it, because in this edge case the user has the possibility to react like @Milena-Czierlinski described before.

Reply:
> This would be a rejection of the deletion process

You can reject a DeleteAttributeRequestItem. However, this would rather be a request to withdraw a formerly accepted Attribute deletion request. I guess that is also what you mean. And for the time being I agree with you that we probably shouldn't implement this.

Reply:
Yes this is what I meant.

Reply:
As discussed we'll allow to re-share an Attribute that is to be deleted by the peer, since simply removing the DeletionInfo via Notification could lead to the Attribute being deleted by the peer after all if they don't actively respond that they won't delete the Attribute and this process is already triggered by their external system. Since this will probably be an edge case, we'll solve this like this instead of implementing another RequestItem to withdraw the accepted deletion of an Attribute for now.